sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/state/ViewportState.kt

```diff
@@ -0,0 +1,50 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import com.mapbox.maps.plugin.viewport.ViewportPlugin
+import com.mapbox.maps.plugin.viewport.experimental.data.ViewportDataSource
+import com.mapbox.maps.plugin.viewport.experimental.transition.TransitionEndListener
+import com.mapbox.maps.plugin.viewport.experimental.transition.ViewportTransitionOptions
+
+/**
+ * Set of possible [ViewportPlugin] states.
+ */
+interface ViewportState {
+  /**
+   * The unique ID of the viewport state.
+   */
+  val id: String
```

Comment:
Not a fan of generic ID based setup if we can avoid them. Since each state has to be registered before being used, can we use the instance reference as the unique key?

Reply:
The ViewportAPI is now refactored to use State instances instead and the generic ID is removed.